app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -0,0 +1,200 @@
+  ORIGIN_METAMASK,
+} from '../../../../shared/constants/app';
+
+export type SubmitSmartTransactionRequest = {
+  transactionMeta: TransactionMeta;
+  smartTransactionsController: SmartTransactionsController;
+  transactionController: TransactionController;
+  isSmartTransaction: boolean;
+  controllerMessenger: any;
+};
+
+export async function submitSmartTransactionHook(
+  request: SubmitSmartTransactionRequest,
+) {
+  const {
+    transactionMeta,
+    smartTransactionsController,
+    transactionController,
+    isSmartTransaction,
+    controllerMessenger,
+  } = request;
+  log.info('Smart Transaction - Executing publish hook', transactionMeta);
+  const isDapp = transactionMeta?.origin !== ORIGIN_METAMASK;
+  const { chainId, txParams } = transactionMeta;
+  // Will cause TransactionController to publish to the RPC provider as normal.
+  const useRegularTransactionSubmit = { transactionHash: undefined };
+  if (!isSmartTransaction) {
+    log.info(
+      `Smart Transaction - Skipping hook as it is not a smart transaction on chainId: ${chainId}`,
+    );
+    return useRegularTransactionSubmit;
+  }
+  let smartTransactionStatusApprovalId: string | undefined;
+  if (isDapp) {
+    const { id } = controllerMessenger.call('ApprovalController:startFlow');
+    smartTransactionStatusApprovalId = id;
+  }
+  try {
+    const feesResponse = await smartTransactionsController.getFees(
+      { ...txParams, chainId },
+      undefined,
+    );
+    log.info('Smart Transaction - Retrieved fees', feesResponse);
+    const signedTransactions = await createSignedTransactions(
+      txParams,
+      feesResponse.tradeTxFees?.fees ?? [],
+      false,
+      transactionController,
+    );
+    const signedCanceledTransactions = await createSignedTransactions(
+      txParams,
+      feesResponse.tradeTxFees?.cancelFees || [],
+      true,
+      transactionController,
+    );
+    log.info('Smart Transaction - Generated signed transactions', {
+      signedTransactions,
+      signedCanceledTransactions,
+    });
+    log.info('Smart Transaction - Submitting signed transactions');
+    const response = await smartTransactionsController.submitSignedTransactions(
+      {
+        signedTransactions,
+        signedCanceledTransactions,
+        txParams,
+        // Patched into controller to skip unnecessary call to confirmExternalTransaction.
+        skipConfirm: true,
+      },
+    );
+    const uuid = response?.uuid;
+    if (!uuid) {
+      throw new Error('No smart transaction UUID');
+    }
+    log.info('Smart Transaction - Received UUID', uuid);
+    if (isDapp) {
+      controllerMessenger.call(
+        'ApprovalController:addRequest',
+        {
+          id: smartTransactionStatusApprovalId,
+          origin,
+          type: SMART_TRANSACTION_CONFIRMATION_TYPES.showSmartTransactionStatusPage,
+          requestState: {
+            smartTransaction: {
+              status: 'pending',
+              creationTime: Date.now(),
```

Comment:
So, on the SmartTransaction type we have `time?: number;`, which I didn't like, because it's not really clear what `time` means. I will add `creationTime` on the SmartTransaction type.